src/ext/framework/mod.rs

```diff
@@ -417,46 +468,4 @@ impl Framework {
 
         self
     }
-
-    /// Adds a "check" to a command, which checks whether or not the command's
-    /// associated function should be called.
-    ///
-    /// # Examples
-    ///
-    /// Ensure that the user who created a message, calling a "ping" command,
-    /// is the owner.
-    ///
-    /// ```rust,no_run
-    /// use serenity::client::{Client, Context};
-    /// use serenity::model::Message;
-    /// use std::env;
-    ///
-    /// let mut client = Client::login_bot(&env::var("DISCORD_TOKEN").unwrap());
-    ///
-    /// client.with_framework(|f| f
-    ///     .configure(|c| c.prefix("~"))
-    ///     .on("ping", ping)
-    ///     .set_check("ping", owner_check));
-    ///
-    /// fn ping(context: &Context, _message: &Message, _args: Vec<String>) {
-    ///     context.say("Pong!");
-    /// }
-    ///
-    /// fn owner_check(_context: &Context, message: &Message) -> bool {
-    ///     // replace with your user ID
-    ///     message.author.id == 7
-    /// }
-    /// ```
-    #[deprecated(since="0.1.2", note="Use the `CreateCommand` builder's `check` instead.")]
-    pub fn set_check<F, S>(mut self, command: S, check: F) -> Self
-        where F: Fn(&Context, &Message) -> bool + Send + Sync + 'static,
-              S: Into<String> {
-        if let Some(command) = self.commands.get_mut(&command.into()) {
-            if let Some(c) = Arc::get_mut(command) {
-                c.checks.push(Box::new(check));
-            }
-        }
-
-        self
-    }
```

Comment:
Although I don't think semver requires this for versions < 1.0.0, it'd be nice to remove this only once we hit v0.2.0, rather than the v0.1.2 that the next version is planned to be. Is it possible to keep this?